examples/todoapp/bin/server/controllers/todos_controller.dart

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2023, Prime Holding JSC
+// https://www.primeholding.com
+//
+// Use of this source code is governed by an MIT-style
+// license that can be found in the LICENSE file or at
+// https://opensource.org/licenses/MIT.
+
+import 'package:shelf/shelf.dart';
+import 'package:shelf_router/shelf_router.dart';
+import 'package:todoapp/base/models/todo_model.dart';
+
+import '../services/todos_service.dart';
+import '../utils/api_controller.dart';
+import '../utils/server_exceptions.dart';
+
+class TodosController extends ApiController {
+  TodosController(this.todosService);
+
+  final TodosService todosService;
+
+  @override
+  void registerRequests(WrappedRouter router) {
+    router.addRequest(
+      RequestType.GET,
+      '/api/todos',
+      getAllTodosHandler,
+    );
+    router.addRequest(
+      RequestType.POST,
+      '/api/todos',
+      addOrUpdateTodoHandler,
```

Comment:
The post method should perform only the create opeation. Feel free to use the PUT method for updating purposes